plugins/BEdita/Core/src/Model/Behavior/HistoryBehavior.php

```diff
@@ -0,0 +1,194 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2019 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\Core\Model\Behavior;
+
+use BEdita\Core\State\CurrentApplication;
+use BEdita\Core\Utility\LoggedUser;
+use Cake\Core\Configure;
+use Cake\Datasource\EntityInterface;
+use Cake\Event\Event;
+use Cake\I18n\Time;
+use Cake\ORM\Behavior;
+use Cake\ORM\TableRegistry;
+
+/**
+ * History behavior
+ * Add history events on object modifications
+ *
+ * @since 4.1.0
+ */
+class HistoryBehavior extends Behavior
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected $_defaultConfig = [
+        'table' => 'History',
+        'exclude' => [
+            'id',
+            'type',
+            'confirm-password',
+        ],
+        'obfuscate' => [
+            'password' => '*****',
+        ],
+        'resource_type' => 'objects',
+    ];
+
+    /**
+     *  History table
+     *
+     * @var \Cake\ORM\Table
+     */
+    public $Table = null;
+
+    /**
+     * The changed properties.
+     * Array of visible properties changed by user
+     *
+     * @var array
+     */
+    public $changed = [];
```

Comment:
I think that define `$changed` as `protected` would be better to avoid direct write on that property. I would expose it with a getter.

Reply:
Ok, done!